### PR TITLE
Merge 1.17 into development

### PR DESCRIPTION
**Pull request by Paldiu**

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/command/impl/SpawnMobCMD.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
-package me.totalfreedom.totalfreedommod.command;
+package me.totalfreedom.totalfreedommod.command.impl;
 
-import java.util.List;
-
-import me.totalfreedom.totalfreedommod.config.ConfigEntry;
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/command/impl/SpawnMobCMD.java:5]: Please re-add the import

**Review by Focusvity — CHANGES_REQUESTED** [review]
I don't know why the formatting has gone to Google-style; maybe reformat the project back to Allman style.

**Comment by Focusvity** [comment]
Because we're no longer supporting Java 11 (by force), we should remove the workflow for Java 11 (as they will fail). @Paldiu @Wild1145 thoughts?

**Comment by Wild1145** [comment]
@Focusvity Once this has been merged we will disable the Java 11 build checks. I'm happy to override the merge on this occasion, however Codacy is un-happy with the code quality for this PR... That needs to be fixed as a matter of absolute urgency.

File: pom.xml [hunk 2]
```diff
@@ -434,6 +441,7 @@
                                     <include>org.jetbrains:annotations</include>
                                 </includes>
                             </artifactSet>
+                            <outputDirectory>E:\A Folder for More Folders\Minecraft Servers\TotalFreedom Testing\plugins</outputDirectory>
```
Comment (Wild1145) [inline pom.xml:444]: This shouldn't have been added

File: pom.xml [hunk 3]
```diff
@@ -401,7 +408,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.2.4</version>
+                <version>3.3.1-SNAPSHOT</version>
```
Comment (Wild1145) [inline pom.xml:411]: We should not be using snapshot versions without there being a good reason, and that reason should be being documented here as a comment. (In the POM, Not Github comments)

File: pom.xml [hunk 4]
```diff
@@ -258,6 +258,13 @@
         </dependency>
     </dependencies>
 
+    <pluginRepositories>
```
Comment (Wild1145) [inline pom.xml:261]: This should not be being added, we should either mirror via Nexus, or not be needing them. In this case again, Snapshots = Bad

File: src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java [hunk 5]
```diff
@@ -296,83 +325,100 @@ public Initializer()
...
+            pvpBlocker = new PVPBlocker();
+            fuckoff = new Fuckoff();
+            autoKick = new AutoKick();
+            autoEject = new AutoEject();
+            monitors = new Monitors();
         }
 
         private void initBridges()
         {
             // Start bridges
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java:383]: We don't seem to tell the server to load these plugins prior to TFM, we should be doing it if we're going to intentionally not load up aspects of TFM based on them running / not.

File: src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java [hunk 6]
```diff
@@ -296,83 +325,100 @@ public Initializer()
...
+            if (isPluginPresent("BukkitTelnet"))
+                bukkitTelnetBridge = new BukkitTelnetBridge();
+            if (isPluginPresent("CoreProtect"))
+                coreProtectBridge = new CoreProtectBridge();
+            if (isPluginPresent("WorldGuard"))
+                worldGuardBridge = new WorldGuardBridge();
+            if (isPluginPresent("WorldEdit"))
+                worldEditBridge = new WorldEditBridge();
+
+            essentialsBridge = new EssentialsBridge();
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java:393]: Why aren't these included in the if logic? Not sure having half of them enabled and half not is a good idea. Especially Libs & TFGuilds where they are not "Mandatory" plugins.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
I've not got the entire way through, I'm also not sure why we've made such substantial changes to the way TFM is structured and refactored just about everything as part of a 1.17 upgrade, much less deployed it on to the server without a passing PR given the exceptionally poor code quality flags...

**Comment by Wild1145** [comment]
Out of my own boredom I've stopped manually cranking the Codacy output into this PR, please as a matter of critical urgency get these resolved as well as other review comments, and then we need to actually perform a proper code review.

**Comment by Wild1145** [comment]
This has been replaced by another PR currently also part of the 2022.02 release. Closing as declined.

**Pull request closed without merging**